src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/ApplicationTest.kt

```diff
@@ -1872,295 +1874,426 @@ class ApplicationTest : IntegrationTestBase() {
+            caseManagerUserDetails = Cas1ApplicationUserDetails(
+              "cmName",
+              "cmEmail",
+              "cmTelephone",
+            ),
+            reasonForShortNotice = "reasonForShort",
+            reasonForShortNoticeOther = "reasonForShortOther",
+          ),
+        )
+        .exchange()
+        .expectStatus()
+        .isOk
+
+      val persistedApplication = approvedPremisesApplicationRepository.findByIdOrNull(applicationId)!!
+
+      assertThat(persistedApplication.noticeType).isEqualTo(Cas1ApplicationTimelinessCategory.standard)
+      assertThat(persistedApplication.isWomensApplication).isFalse()
+      assertThat(persistedApplication.isPipeApplication).isFalse
+      assertThat(persistedApplication.targetLocation).isEqualTo("SW1A 1AA")
+      assertThat(persistedApplication.sentenceType).isEqualTo(SentenceTypeOption.nonStatutory.toString())
+      assertThat(persistedApplication.apArea?.id).isEqualTo(submittingUser.apArea!!.id)
+
+      val createdAssessment =
+        approvedPremisesAssessmentRepository.findAll().first { it.application.id == applicationId }
+      assertThat(createdAssessment.allocatedToUser).isNull()
+      assertThat(createdAssessment.createdFromAppeal).isFalse()
+
+      val persistedDomainEvents = domainEventRepository.findAll().filter { it.applicationId == applicationId }
+      assertThat(persistedDomainEvents).hasSize(1)
+      val persistedApplicationSubmittedEvent = persistedDomainEvents.firstOrNull { it.type == DomainEventType.APPROVED_PREMISES_APPLICATION_SUBMITTED }
+
+      assertThat(persistedApplicationSubmittedEvent).isNotNull
+      assertThat(persistedApplicationSubmittedEvent!!.crn).isEqualTo(offenderDetails.otherIds.crn)
+
+      val emittedMessage = domainEventAsserter.blockForEmittedDomainEvent(DomainEventType.APPROVED_PREMISES_APPLICATION_SUBMITTED)
+
+      val emittedMessageDescription = "An application has been submitted for an Approved Premises placement"
+      assertThat(emittedMessage.description).isEqualTo(emittedMessageDescription)
+      assertThat(emittedMessage.detailUrl).matches("http://api/events/application-submitted/[0-9a-fA-F]{8}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{12}")
+      assertThat(emittedMessage.additionalInformation.applicationId).isEqualTo(applicationId)
+      assertThat(emittedMessage.personReference.identifiers).containsExactlyInAnyOrder(
+        SnsEventPersonReference("CRN", offenderDetails.otherIds.crn),
+        SnsEventPersonReference("NOMS", offenderDetails.otherIds.nomsNumber!!),
+      )
+
+      emailAsserter.assertEmailsRequestedCount(1)
+      emailAsserter.assertEmailRequested(
+        toEmailAddress = submittingUser.email!!,
+        templateId = notifyConfig.templates.applicationSubmitted,
+        replyToEmailId = persistedApplication.apArea!!.notifyReplyToEmailId,
+      )
+    }
+
+    @Test
+    fun `Submit emergency application auto allocates the assessment`() {
```

Comment:
Possibly change test name to 'Submit emergency application auto allocates the assessment sends emails and raises domain events'()